src/check/parse/IR.zig

```diff
@@ -1678,42 +1678,97 @@ pub const NodeStore = struct {
                         env.gpa,
                         ir.resolve(import.module_name_tok),
                     );
 
                     // Module Alias e.g. `OUT` in `import pf.Stdout as OUT`
-                    import_node.appendStringChild(
+                    node.appendStringChild(
                         env.gpa,
                         if (import.alias_tok) |tok| ir.resolve(tok) else "",
                     );
 
                     // Each exposed identifier e.g. [foo, bar] in `import pf.Stdout exposing [foo, bar]`
                     for (ir.store.tokenSlice(import.exposes)) |tok| {
-                        import_node.appendStringChild(env.gpa, ir.resolve(tok));
+                        node.appendStringChild(env.gpa, ir.resolve(tok));
                     }
 
-                    return import_node;
+                    return node;
                 },
-                else => {
-                    std.log.err("format for statement {}", .{self});
-                    @panic("not implemented");
+                // (type_decl (header <name> [<args>]) <annotation>)
+                .type_decl => |a| {
+                    var node = sexpr.Expr.init(env.gpa, "type_decl");
+                    var header = sexpr.Expr.init(env.gpa, "header");
+
+                    // pattern
+                    {
+                        const ty_header = ir.store.getTypeHeader(a.header);
+
+                        header.appendStringChild(env.gpa, ir.resolve(ty_header.name));
+
+                        for (ir.store.tokenSlice(ty_header.args)) |b| {
+                            header.appendStringChild(env.gpa, ir.resolve(b));
+                        }
+
+                        node.appendNodeChild(env.gpa, &header);
+                    }
+                    // annotation
+                    {
+                        var annotation = ir.store.getTypeAnno(a.anno).toSExpr(env, ir);
+                        node.appendNodeChild(env.gpa, &annotation);
+                    }
+
+                    return node;
+                },
+                .crash => {
+                    const node = sexpr.Expr.init(env.gpa, "crash");
+                    // : struct {
+                    //     expr: ExprIdx,
+                    //     region: Region,
+                    // },
+
+                    return node;
+                },
+                .expect => {
+                    const node = sexpr.Expr.init(env.gpa, "expect");
+                    // : struct {
+                    //     body: ExprIdx,
+                    //     region: Region,
+                    // },
+                    return node;
+                },
+                .@"return" => {
+                    const node = sexpr.Expr.init(env.gpa, "return");
+                    // : struct {
+                    //     expr: ExprIdx,
+                    //     region: Region,
+                    // },
+                    return node;
```

Comment:
These are all missing the child expressions.

Reply:
hmmm. Yeah I removed the `else` branch... and at the time was thinking I'd do these in a follow up and then forgot about them. 

But I've just grabbed a coffee... so I'll finish these now 😃